src/type_checker/mod.rs

```diff
@@ -333,16 +333,20 @@ impl<B: Backend> TypeChecker<B> {
                     };
 
                     // store generic parameters as local vars in the fn_env
-                    for gen in &function.sig.generics {
+                    for name in function.sig.generics.names() {
                         typed_fn_env.store_type(
-                            gen.to_string(),
-                            // todo: let generic vars carry their span
-                            TypeInfo::new(TyKind::Generic(gen.to_string()), function.span),
+                            name.to_string(),
+                            TypeInfo::new_cst(TyKind::Field, function.span),
```

Comment:
`name.span`, the generics should be `Ident`

Reply:
The generics store parameters by their unique names.

If the hash map is based on the `Ident`, it can be difficult to point to the right one?
such as 
```rust
fn arr(aa: [[Field; N]; N])
```

Reply:
oh I see, good point!